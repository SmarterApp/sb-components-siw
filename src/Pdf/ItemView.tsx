import * as React from "react";
import * as Path from "path";
import * as os from "os";
import { ItemPdfModel, PdfViewType } from "./PdfModels";

export interface ItemViewProps {
  view: ItemPdfModel;
}

export class ItemView extends React.Component<ItemViewProps, {}> {
  render() {
    let item: JSX.Element | undefined;
    const { view } = this.props;
    if (view.type === PdfViewType.html) {
      // tslint:disable-next-line:react-no-dangerous-html
      item = <div dangerouslySetInnerHTML={{ __html: view.html || "" }} />;
    } else if (view.screenshotUrl) {
      item = <img alt="" role="presentation" src={view.screenshotUrl} />;
    }

    return item;
  }
}
