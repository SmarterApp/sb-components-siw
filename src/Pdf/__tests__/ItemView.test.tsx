import * as React from "react";
import * as ReactDOM from "react-dom";
import { ItemView } from "../ItemView";
import { ItemPdfModel, PdfViewType } from "../PdfModels";
import { shallow, mount, render } from "enzyme";

const itemView: ItemPdfModel = {
  id: "12334",
  html: undefined,
  picturePath: undefined,
  captured: true,
  type: PdfViewType.html
};

const itemView1: ItemPdfModel = {
  id: "12334",
  html: "<div>hello world</div>",
  picturePath: undefined,
  captured: true,
  type: PdfViewType.html
};

const itemView2: ItemPdfModel = {
  id: "12334",
  html: "<div>hello world</div>",
  picturePath: "filename",
  captured: true,
  type: PdfViewType.picture
};

describe("EvidenceStatement", () => {
  it("matches snapshot", () => {
    expect(shallow(<ItemView view={itemView} />)).toMatchSnapshot();
  });

  it("matches snapshot with html", () => {
    expect(shallow(<ItemView view={itemView1} />)).toMatchSnapshot();
  });

  it("matches snapshot with picture", () => {
    expect(shallow(<ItemView view={itemView2} />)).toMatchSnapshot();
  });
});
