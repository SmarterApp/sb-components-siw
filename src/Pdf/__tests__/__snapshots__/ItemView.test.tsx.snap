// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvidenceStatement matches snapshot 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "",
    }
  }
/>
`;

exports[`EvidenceStatement matches snapshot with html 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<div>hello world</div>",
    }
  }
/>
`;

exports[`EvidenceStatement matches snapshot with picture 1`] = `""`;
